src/decorators/factory.ts

```diff
@@ -0,0 +1,28 @@
+import {DecoratorFactory, GenericDecorator, DecoratorConfig} from 'lodash-decorators';
+import {Applicator, ApplicateOptions} from 'lodash-decorators/applicators';
+
+/**
+ * A simplified version of a method decorator
+ */
+export interface SimpleMethodDecorator {
+  (method: Function, thisArg: object, ...decoratorArgs: any[]): any,
+}
+
+class GenericApplicator extends Applicator {
```

Comment:
I'm sure this pays dividends in a large library of decorators, but it feels like mind-melting overkill for our use case

Reply:
I think we have a bunch of alternative options here:

1. I can move this code into `react-decorators` where it's used, instead of making a factory for decorators, and just make the one decorator there.

2. I can create a `callback` decorator in `react-utilities` and we can document that it won't work together with other decorators (it probably won't need to, except maybe debounce, but we can cross that bridge when we come to it).

3. We can can the decorator idea entirely, and create a function `memoizeCallback(fn, this,  args)` to handle our use case.

My preference is option `1`, because it eases the transition to the current solution, should we need it for other decorators later on.

Reply:
On further reflection, I think that's a bad reason for choosing option `1`. What is your preference? I'm open to either `2` or `3`.